### ISSUE TITLE
add option for NVDA to not automatically read first line of an email in outlook

### ISSUE THREAD

**Issue by Dennisl123** [nvaccess/nvda#15310]
Hi, please add an option for NVDA to not read the first line of an email message automatically.  This is a problem because not all user want this.  User control should be what matters here.  I don't want this. More over other screen readers have this option and had this option.  Jaws and windoweyes come to mind.  This also ignores the auto read function of automatically say all on browse mode is turned off it shouldn't read any part of the message.  Please add this option asap.

**Comment by Adriani90** [comment]
See also #4977 for a systematic approach which is requested for visual readers.
In the means of completeness, at least for browse mode this setting could be implemented as part of solving #3020 without having to add a new additional option.
The setting "Automatic say all on page load" is buggy in Outlook since it doesn't work as expected. When this setting is turned off, NVDA should not read anything from the message body when opening the email in browse mode.

An additional setting is only needed when opening the email message in focus mode. But this could be done when implementing #4977.

cc: @michaelDCurran I think a decision is needed from NV Access side on how to further proceed with this.

**Comment by amirsol81** [comment]
I also support this feature. @Qchristensen

**Comment by Dennisl123** [comment]
I hope to see these issues fixed sooner then later.  Any idea when they will be fixed?

**Comment by joshknnd1982** [comment]
I support this feature also. But I would like to add: at least have NVDA play a sound to let the user know when the message is fully loaded into the browse mode buffer and is ready to be read.

**Comment by Dennisl123** [comment]
If that sound option is added it should be able to be turned off if people don’t want the sound to play.

**Comment by joshknnd1982** [comment]
Other screen readers have this feature. Window-eyes had it 30 years ago. Many screen reader users find this useful.

**Comment by Dennisl123** [comment]
Hi, this is a long standing issue.  Can this please be fixed or added?  Thanks.

**Comment by Dennisl123** [comment]
if automatic say all is turned off this doesn't stop the first line from automatically being read.  Please add an option to prevent this.  Thank you.

**Comment by SaschaCowley** [comment]
I am a little perplexed by this issue. NVDA reads the current line of a browse mode document when the document is focused, regardless of the value of Automatic Say All on page load. So the current behaviour in Outlook mirrors that in Thunderbird, Firefox, Edge, Chrome, and likely other applications. What would you have NVDA vocalise instead? And why should this setting be restricted to (presumably classic) Outlook?

**Comment by joshknnd1982** [comment]
Dennis wants NVDA to optionally behave  the way jaws does. So when an email is opened up in outlook for the first time, and NVDA goes into browse mode, NVDA should just play a short beep to let you know that the message is loaded and that it is able to be read. Ideally, he wants this to only happen in modern outlook and classic outlook, or if it is easier, implement this as an optional browse mode configuration in settings. In other words, there should be three options for "automatic say all in browse mode." off, on, or beep. When set to beep or play sound, NVDA will just play a sound and say nothing once the browse mode buffer is loaded and the page can now be read and interacted with.

**Comment by Dennisl123** [comment]
This is what I am expecting to happen.  Note I am asking this to be an option.
This is an option Jaws has. I expect if it is unchecked in NVDA it will not read anything including the first line.
Messages Automatically Read not checked
If this is unchecked it reads nothing when a message is opened.

**Comment by joshknnd1982** [comment]
This is an option Jaws has.  I expect if it is unchecked in NVDA it will not read anything including the first line.
Messages Automatically Read not checked

**Comment by SaschaCowley** [comment]
I doubt we would accept a PR adding this exclusively to Outlook. Adding it to Browse Mode in general, you could have an Outlook profile with this feature enabled to restrict this option to Outlook.

**Comment by Dennisl123** [comment]
that would be fine.  when do you think this  will be added? Dennis Long Sent from my iPhone 16EOn Jul 24, 2025, at 10:38 PM, Sascha Cowley ***@***.***> wrote:﻿SaschaCowley left a comment (nvaccess/nvda#15310)
I doubt we would accept a PR adding this exclusively to Outlook. Adding it to Browse Mode in general, you could have an Outlook profile with this feature enabled to restrict this option to Outlook.

**Comment by SaschaCowley** [comment]
@Dennisl123 I don't know. This is not something we're likely to implement internally, but if a community member wants to work on it they are more than welcome.

**Comment by CyrilleB79** [comment]
@Dennisl123, @joshknnd1982 it's not all clear to me.

Could you indicate:
1. How would the new option would be named and where it should land, in which category. The name would not be definitive but it would help understand what it does.
2. Could you describe what happens in the 4 following cases when opening an e-mail, when alt+tabbing back to an already opened e-mail, when opening a new page in a browser or when alt+tabbing to an already opened page?
   * Case 1.1: "Automatic Say All on page load" on and new option on
   * Case 1.2: "Automatic Say All on page load" on and new option off
   * Case 2.1: "Automatic Say All on page load" off and new option on
   * Case 2.2: "Automatic Say All on page load" off and new option off

This would help progress this issue. Thanks.

**Comment by joshknnd1982** [comment]
answers are below the questions...

1. How would the new option would be named and where it should land, in which category. The name would not be definitive but it would help understand what it does.
answer: the new option is named automatic say all on page load, it would be a combo box. It is in settings, under browse mode.  there are three choices, on, off, and beep.

2. Could you describe what happens in the 4 following cases 
when opening an e-mail, 
answer: NVDA will load the email as normal into the browse mode buffer, when set to on, NVDA starts reading as normal, as if it were a checkbox that was checked in the old automatic say all choices, in other words it is read as normal, when set to off, the first line is read, and when set to beep, you just hear a beep sound and nothing is read but you know that you are free to read when you wish by pressing a keyboard shortcut, nothing is spoken by the tts until a command is issued
when alt+tabbing back to an already opened e-mail, 
answer: if set to on, NVDA reads the whole browse mode buffer as if the checkbox were checked in the automatic say all checkbox, if set to off, the first line is read, if set to beep, NVDA beeps to notify you that all is well and you can now press keyboard shortcuts to read the email, but nothing is spoken through the tts 
when opening a new page in a browser or when alt+tabbing to an already opened page?
answer: when set to on, NVDA reads the entire browse mode buffer like it does now, when set to off, the first line is read and then NVDA stops, when set to beep, nothing is spoken, but NVDA beeps to let you know you can now read by pressing a command of your choice

◦ Case 1.1: "Automatic Say All on page load" on and new option on
answer: NVDA behaves as normal, just like when the checkbox is checked under browse mode options in settings
◦ Case 1.2: "Automatic Say All on page load" on and new option off
answer: the first line of the browse mode buffer is read, like when the checkbox is unchecked
◦ Case 2.1: "Automatic Say All on page load" off and new option on
answer: when the new beep option is on, nothing is read through the tts, you just hear a beep to let you know you can now read the document when you choose

◦ Case 2.2: "Automatic Say All on page load" off and new option off
answer: if off is chosen, the first line of the browse mode buffer is read, then NVDA stops reading

**Comment by Dennisl123** [comment]
No if set to off the first line is not read!

**Comment by Dennisl123** [comment]
If set to off and you alt tab the subject is read.  That is it.

**Comment by joshknnd1982** [comment]
ok, did you comment on it with your correction?

**Comment by joshknnd1982** [comment]
yes, NVDA should just read the message subject when alt tabbing back into an outlook email message, and nothing more.

**Comment by CyrilleB79** [comment]
OK. So if I understand correctly you both suggest to add a new option in the existing option instead of creating a new option as I have thought before.

The new option could be renamed just "On page load" instead of "Automatic Say All on page load":
And the 3 values would be:
* "Start reading the page" (identical as when the old option was checked)
* "Make a beep" (we may find a better wording later)
* For the third option, you disagree one with another:
  * "Do nothing" (suggestion from @Dennisl123 )
  * "Read the current line" (suggestion from @joshknnd1982)

It's worth noting that in @Dennisl123's suggestion, there is no equivalent of the behaviour when the old option was set to off (unchecked). I guess it's a problem for people who

**Comment by Dennisl123** [comment]
I would like it to do nothing.  Maybe add a forth option for those that want it to read the first line.

**Comment by Dennisl123** [comment]
Do you have any idea when this will be added?

**Comment by Dennisl123** [comment]
do we know when this will be added?

**Comment by joshknnd1982** [comment]
Will this be added soon? This is a longstanding issue.
